system/training.py

```diff
@@ -91,125 +98,168 @@ def filter_groups(param_groups, groups):
         combined_conf,
         combined_masses,
         combined_dp_idxs,
-        1000,
-        None
+        1000
     )
-
-    HG_E, HG_derivs, _ = simulation.average_E_and_derivatives(RHG) # [combined_dp_idxs,]
     
+    HG_E, HG_derivs, _ = simulation.average_E_and_derivatives(RHG)
+        
     pred_enthalpy = HG_E - (G_E + H_E)
     delta = pred_enthalpy - label
-    num_atoms = len(combined_masses)
-    loss = delta**2
     
-    # fancy index into the full derivative set
     combined_derivs = np.zeros_like(combined_params)
-    # remember its HG - H - G
     combined_derivs[combined_dp_idxs] += HG_derivs
     combined_derivs[host_dp_idxs] -= H_derivs
     combined_derivs[guest_dp_idxs + len(host_params)] -= G_derivs
-    # combined_derivs = 2*delta*combined_derivs # L2 derivative
-    combined_derivs = (delta/np.abs(delta))*combined_derivs # L1 derivative
+
+    if properties['loss_fn'] == 'L2':
+        combined_derivs = 2*delta*combined_derivs
+    elif properties['loss_fn'] == 'L1':
+        combined_derivs = (delta/np.abs(delta))*combined_derivs
+    elif properties['loss_fn'] == 'Huber':
+        # Modified Huber Loss
```

Comment:
Out of curiosity, did Huber loss perform better than L1/L2?

Reply:
Depended on the system. When I was doing absolute energies with the fully minimized host-guests, Huber performed better than L1 & L2, but L2 has seemed to work fine so far for relative fitting